docs/runtime-contract.md

```diff
@@ -162,7 +162,7 @@ A read from the `stdin` file descriptor on the container SHOULD always result in
 collected and retained in a developer-accessible logging repository.
 (TODO:[docs#902](https://github.com/knative/docs/issues/902)).
 
-Within the container, pipes and file descriptors may be used to communicate
+Within the container, pipes and file descriptors MAY be used to communicate
```

Comment:
The MAY keyword doesn't seem appropriate here. If a container MAY be able to do this it implies that the platform MUST allow pipes and file descriptors for IPC. Using MUST and phrasing from the platform point-of-view seems like the more appropriate wording, but the statement seems like overkill.

I cannot think of a container runtime that doesn't allow this and I cannot think of a good reason for a future runtime to disable this.

@evankanderson Any thoughts on why this is here?

Reply:
changing it to a "can" since we have no say over what happens within the container w.r.t. these pipes - IOW, I think this statement is out of scope of Knative.

Reply:
If we are going to change to "can" I would prefer removal of it. Keeping this document concise keeps it readable and usable. I don't think it clarifies much for platform providers or users as a "can" statement.

Reply:
I think this was added at one point due to an internal discussion about required gvisor capabilities. In any case, it was a point-in-time thing, and I agree with removing it, unless we're going to spell out a complete set of linux syscalls which are in-scope.